From: Ben Hutchings <ben@decadent.org.uk>
Date: Fri, 27 Jan 2012 03:41:01 +0000
Subject: [PATCH] Revert "fix cputime overflow in uptime_proc_show"

This reverts commit bbe520ae8b0042af8bf4cebe61a860b318f40314, which
conflicts with VServer changes to calculation of idle time.
---
 fs/proc/uptime.c |    9 ++-------
 1 files changed, 2 insertions(+), 7 deletions(-)

diff --git a/fs/proc/uptime.c b/fs/proc/uptime.c
index 29166ec..766b1d4 100644
--- a/fs/proc/uptime.c
+++ b/fs/proc/uptime.c
@@ -11,20 +11,15 @@ static int uptime_proc_show(struct seq_file *m, void *v)
 {
 	struct timespec uptime;
 	struct timespec idle;
-	cputime64_t idletime;
-	u64 nsec;
-	u32 rem;
 	int i;
+	cputime_t idletime = cputime_zero;
 
-	idletime = 0;
 	for_each_possible_cpu(i)
 		idletime = cputime64_add(idletime, kstat_cpu(i).cpustat.idle);
 
 	do_posix_clock_monotonic_gettime(&uptime);
 	monotonic_to_bootbased(&uptime);
-	nsec = cputime64_to_jiffies64(idletime) * TICK_NSEC;
-	idle.tv_sec = div_u64_rem(nsec, NSEC_PER_SEC, &rem);
-	idle.tv_nsec = rem;
+	cputime_to_timespec(idletime, &idle);
 	seq_printf(m, "%lu.%02lu %lu.%02lu\n",
 			(unsigned long) uptime.tv_sec,
 			(uptime.tv_nsec / (NSEC_PER_SEC / 100)),
-- 
1.7.8.2

